test-ref-configs.pl fails with Clang

### Description
- Type: Bug
- Priority: Major

---------------------------------------------------------------
## Bug

**mbed TLS build:**  
Version: `mbedtls-2.13.1`

Executing `test-ref-configs.pl` with clang fails with a series of build errors.

Extract from the build.

```
  CC    x509/req_app.c
  Gen   test_suite_aes.ofb.c
  CC    test_suite_aes.ofb.c
suites/helpers.function:238:13: error: unused function 'hexify'
      [-Werror,-Wunused-function]
static void hexify( unsigned char *obuf, const unsigned char *ibuf, int len )
            ^
suites/helpers.function:292:23: error: unused function 'unhexify_alloc'
      [-Werror,-Wunused-function]
static unsigned char *unhexify_alloc( const char *ibuf, size_t *olen )
                      ^
suites/helpers.function:343:12: error: unused function 'rnd_zero_rand'
      [-Werror,-Wunused-function]
static int rnd_zero_rand( void *rng_state, unsigned char *output, size_t len )
           ^
suites/helpers.function:370:12: error: unused function 'rnd_buffer_rand'
      [-Werror,-Wunused-function]
static int rnd_buffer_rand( void *rng_state, unsigned char *output, size_t len )
           ^
suites/helpers.function:416:12: error: unused function 'rnd_pseudo_rand'
      [-Werror,-Wunused-function]
static int rnd_pseudo_rand( void *rng_state, unsigned char *output, size_t len )
           ^
5 errors generated.
Makefile:102: recipe for target 'test_suite_aes.ofb' failed
make[1]: *** [test_suite_aes.ofb] Error 1
Makefile:21: recipe for target 'tests' failed
make: *** [tests] Error 2
Failed to build: config-suite-b.h
```

--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,10 @@
 mbed TLS ChangeLog (Sorted per branch, date)
 
+= mbed TLS x.x.x branch released xxxx-xx-xx
+
+Bugfix
+   * Fix to allow test-ref-config.pl to build with Clang. Fixes #1628.
+
 = mbed TLS 2.7.9 branch released 2018-12-21
 
 Bugfix

--- a/ChangeLog
+++ b/ChangeLog
@@ -3,7 +3,8 @@ mbed TLS ChangeLog (Sorted per branch, date)
 = mbed TLS x.x.x branch released xxxx-xx-xx
 
 Bugfix
-   * Fix to allow test-ref-config.pl to build with Clang. Fixes #1628.
+   * Fix to allow building test suites with any warning that detects unused
+     functions. Fixes #1628.
 
 = mbed TLS 2.7.9 branch released 2018-12-21
 

--- a/tests/Makefile
+++ b/tests/Makefile
@@ -3,7 +3,7 @@
 # To compile with PKCS11: add "-lpkcs11-helper" to LDFLAGS
 
 CFLAGS	?= -O2
-WARNING_CFLAGS ?= -Wall -W -Wdeclaration-after-statement -Wno-unused-function -Wno-unused-value
+WARNING_CFLAGS ?= -Wall -W -Wdeclaration-after-statement -Wunused
 LDFLAGS ?=
 
 LOCAL_CFLAGS = $(WARNING_CFLAGS) -I../include -D_FILE_OFFSET_BITS=64

--- a/tests/suites/helpers.function
+++ b/tests/suites/helpers.function
@@ -128,6 +128,14 @@ test_info;
 /*----------------------------------------------------------------------------*/
 /* Helper Functions */
 
+void test_fail( const char *test, int line_no, const char* filename )
+{
+    test_info.failed = 1;
+    test_info.test = test;
+    test_info.line_no = line_no;
+    test_info.filename = filename;
+}
+
 #if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
 static int redirect_output( FILE** out_stream, const char* path )
 {
@@ -170,7 +178,7 @@ static void close_output( FILE* out_stream )
 }
 #endif /* __unix__ || __APPLE__ __MACH__ */
 
-static int unhexify( unsigned char *obuf, const char *ibuf )
+int unhexify( unsigned char *obuf, const char *ibuf )
 {
     unsigned char c, c2;
     int len = strlen( ibuf ) / 2;
@@ -204,7 +212,7 @@ static int unhexify( unsigned char *obuf, const char *ibuf )
     return len;
 }
 
-static void hexify( unsigned char *obuf, const unsigned char *ibuf, int len )
+void hexify( unsigned char *obuf, const unsigned char *ibuf, int len )
 {
     unsigned char l, h;
 
@@ -258,7 +266,7 @@ static unsigned char *zero_alloc( size_t len )
  *
  * For convenience, dies if allocation fails.
  */
-static unsigned char *unhexify_alloc( const char *ibuf, size_t *olen )
+unsigned char *unhexify_alloc( const char *ibuf, size_t *olen )
 {
     unsigned char *obuf;
 
@@ -309,7 +317,7 @@ static int rnd_std_rand( void *rng_state, unsigned char *output, size_t len )
  *
  * rng_state shall be NULL.
  */
-static int rnd_zero_rand( void *rng_state, unsigned char *output, size_t len )
+int rnd_zero_rand( void *rng_state, unsigned char *output, size_t len )
 {
     if( rng_state != NULL )
         rng_state  = NULL;
@@ -336,7 +344,7 @@ typedef struct
  *
  * After the buffer is empty it will return rand();
  */
-static int rnd_buffer_rand( void *rng_state, unsigned char *output, size_t len )
+int rnd_buffer_rand( void *rng_state, unsigned char *output, size_t len )
 {
     rnd_buf_info *info = (rnd_buf_info *) rng_state;
     size_t use_len;
@@ -382,7 +390,7 @@ typedef struct
  *
  * rng_state shall be a pointer to a rnd_pseudo_info structure.
  */
-static int rnd_pseudo_rand( void *rng_state, unsigned char *output, size_t len )
+int rnd_pseudo_rand( void *rng_state, unsigned char *output, size_t len )
 {
     rnd_pseudo_info *info = (rnd_pseudo_info *) rng_state;
     uint32_t i, *k, sum, delta=0x9E3779B9;

--- a/tests/suites/helpers.function
+++ b/tests/suites/helpers.function
@@ -424,10 +424,3 @@ int rnd_pseudo_rand( void *rng_state, unsigned char *output, size_t len )
     return( 0 );
 }
 
-static void test_fail( const char *test, int line_no, const char* filename )
-{
-    test_info.failed = 1;
-    test_info.test = test;
-    test_info.line_no = line_no;
-    test_info.filename = filename;
-}

